File: 2022/day1/day1.py
if __name__ == '__main__':
    with open('in.txt') as file:
        elves = [list(map(int, elf.split('\n'))) for elf in file.read().split('\n\n')]

    top_elves = sorted(map(sum, elves), reverse=True)[:3]

    print('Part 1:', top_elves[0])
    print('Part 2:', sum(top_elves))

File: 2022/day1/Day1.kt
import java.nio.file.Files
import java.nio.file.Path

fun main() = Files.readString(Path.of("src/main/resources/in.txt"))
    .split("\r\n\r\n")
    .map { it.split("\r\n").map(Integer::parseInt) }
    .map { it.sum() }
    .sortedDescending()
    .take(3)
    .let {
        println("Part 1: ${it.first()}")
        println("Part 2: ${it.sum()}")
    }
